lib/cdo/server_tools.rb

```diff
@@ -0,0 +1,101 @@
+# Automation for manipulating production servers.
+#
+# Requires the aws cli tools.  On OSX these can be installed via:
+# sudo easy_install pip && sudo pip install awscli && aws configure
+
+require 'json'
+require_relative 'newrelic'
+
+class ServerTools
+
+  # Returns the Chef json hash for all of the servers matching `name_glob`.
+  def self.find_frontends(name_glob)
+    JSON.parse(
+      `knife search node "roles:front-end AND chef_environment:production AND name:#{name_glob}" --format json`)['rows']
+  end
+
+  # Returns a [name, hostname, instance_id] tuple for each of the servers whose name matches `name_glob`.
+  def self.find_frontend_names(name_glob)
+    find_frontends(name_glob).map {|fe| [fe['name'], fe['automatic']['ec2']['hostname'], fe['automatic']['ec2']['instance_id']]}
+  end
+
+  # Returns the instance ids of the servers whose name matches `name_glob`.
+  def self.find_frontend_instance_ids(name_glob)
+    find_frontend_names(name_glob).map {|t| t[2]}
+  end
+
+  # Returns the hostnames of the servers whose name matches `name_glob`.
+  def self.find_frontend_hosts(name_glob)
+    find_frontend_names(name_glob).map {|t| t[1]}
+  end
+
+  # Disables New Relic alerts for all servers whose name matches `name_glob`.
+  def self.disable_newrelic_alerts(name_glob)
+    NewRelic.disable_alerts(find_frontend_hosts(name_glob))
+  end
+
+  # Enables production New Relic alerts for all servers whose name matches `name_glob`.
+  def self.enable_newrelic_alerts(name_glob)
+    NewRelic.enable_alerts(find_frontend_hosts(name_glob))
+  end
+
+  # Deregister the frontend instances matching `name_glob` from the production elbs.
+  def self.deregister_frontend_instances(name_glob)
+    approved, names = prompt_for_action('deregister', name_glob)
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `names`. Use `_` or `_names` as a variable name to indicate that it won't be used.